kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaPublisher.java

```diff
@@ -86,7 +88,11 @@ protected KafkaPublisher(Builder<K, V> builder) {
         this.producerFactory = builder.producerFactory;
         this.messageConverter = builder.messageConverter;
         this.messageMonitor = builder.messageMonitor;
-        this.topic = builder.topic;
+        if (builder.topicFunction == null) {
```

Comment:
I think it's a little cleaner to not have this logic in the constructor, but rather in the Builder.
We've done similar things in the `Repositories` in the framework.
Definitely a nit, but wanted to share it regardless.

Reply:
Yes, done